src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -715,6 +715,27 @@ set.subset_set_smul_iff
 
 end pointwise_actions
 
+section
+
+variables {L : Type*} [field L]
+
+/--The pullback of `A` along the ring homomorphism `K →+* L`. -/
+def comap (A : valuation_subring L) (f : K →+* L) :
+  valuation_subring K :=
+{ mem_or_inv_mem' := λ k, by simp [valuation_subring.mem_or_inv_mem],
+  ..(A.to_subring.comap f) }
+
+/--The ring homomorphism from `comap A f` to `A`. -/
+def ring_hom.valuation_subring_comap (A : valuation_subring L) (f : K →+* L) :
+   (comap A f) →+* A :=
+{ to_fun := λ x, ⟨f x.1, x.2⟩,
+  map_one' := subtype.eq f.map_one,
+  map_mul' := λ x y, subtype.eq (f.map_mul x y),
+  map_add' := λ x y, subtype.eq (f.map_add x y),
+  map_zero' := subtype.eq f.map_zero, }
```

Comment:
There should be a general procedure to restrict a `ring_hom` from `R →+* S` to subrings `R' →+* S'` provided `R' ≤ S'.comap f`. 

This should be easily achieved with `ring_hom.restrict` and `ring_hom.cod_restrict`, and should go to the same file and below them. We should probably rename `restrict` to `dom_restrict` or similar, and name the new construction to `ring_hom.restrict` instead.

After this, I think this definition wouldn't be nessecary.

Reply:
OK, thank you very much - I 'll work on this

Reply:
OK, done. I' ve added the new construction. Please let me know how to change the names of these definitions- ( I guess we have to change them in multiple files...?) and what else to do except for removing the `ring_hom.valuation_subring_comap ` Thank you